python/cudf/cudf/_lib/hash.pyx

```diff
@@ -51,32 +51,24 @@ def hash_partition(list source_columns, object columns_to_hash,
 
 @acquire_spill_lock()
 def hash(list source_columns, str method, int seed=0):
-    cdef table_view c_source_view = table_view_from_columns(source_columns)
-    cdef unique_ptr[column] c_result
+    ctbl = pylibcudf.Table([c.to_pylibcudf(mode="read") for c in source_columns])
```

Comment:
*nit*: I think that we can provide type info, which should also us to use the C version of the cpdef functions?

So, up top, `from cudf._lib cimport pylibcudf`

And here:

```
cdef pylibcudf.Table = pylibcudf.Table([...])
```
I think?